scripts/components/custom-select/docs/readme.mdx

```diff
@@ -50,6 +50,27 @@ Your IDE will display all the properties, together with their descriptions.
 />
 ```
 
+### Async single item with modified output
+```jsx
+<CustomSelect
+	loadOptions={getFetchWpApi(getPostTypeEndpoint, {fields: 'id,title,type', processMetadata: ({ type }) => type })}
+	placeholder={__('Select a single item', 'eigthshift-frontend-libs')}
+	value={myValue}
+	onChange={(value) => setAttributes({ myValue: value })}
+	multiple={false}
+	label={__('My label', 'eigthshift-frontend-libs')}
+	filterAsyncOptions={(options) => !Array.isArray(options) ?
+		options :
+		[...options].map(({label, value, metadata}) => {
+			return {
+				label: `${label} -- ${metadata}`,
+				value,
+			};
+		});
+	}
```

Comment:
```suggestion
	filterAsyncOptions={(options) => {
			if (!Array.isArray(options)) {
				return options;
			}

			return [...options].map(({label, value, metadata}) => {
				return {
					label: `${label} -- ${metadata}`,
					value,
				};
			});
		}
	}
```

Reply:
I did it on purpose so it's a bit shorter. But if you insist 🙂